zrml/prediction-markets/src/lib.rs

```diff
@@ -570,123 +545,64 @@ mod pallet {
-            let weight_len = weights.len().saturated_into();
-
-            // Deploy a swap pool with MinLiquidity
-            let _ = Self::deploy_swap_pool_for_market(origin, market_id, weights)?;
-            let pool_id = T::MarketCommons::market_pool(&market_id)?;
-            let mut weight_pool_joins_and_sells = 0;
-            let mut add_liqudity =
-                |amount: BalanceOf<T>, asset: Asset<MarketIdOf<T>>| -> DispatchResult {
-                    let local_weight = T::Swaps::pool_join_with_exact_asset_amount(
-                        who.clone(),
-                        pool_id,
-                        asset,
-                        amount,
-                        zero_balance,
-                    )?;
-                    weight_pool_joins_and_sells =
-                        weight_pool_joins_and_sells.saturating_add(local_weight);
-                    Ok(())
-                };
-
-            // Add additional liquidity as specified in amount_outcome_assets
-            for (idx, asset_amount) in amount_outcome_assets.iter().enumerate() {
-                if *asset_amount == zero_balance {
-                    continue;
-                };
-
-                let remaining_amount =
-                    (*asset_amount).saturating_sub(MinLiquidity::get().saturated_into());
-                let asset_in = match assets {
-                    MarketType::Categorical(_) => {
-                        Asset::CategoricalOutcome(market_id, idx.saturated_into())
-                    }
-                    MarketType::Scalar(_) => {
-                        if idx == 0 {
-                            Asset::ScalarOutcome(market_id, ScalarPosition::Long)
-                        } else {
-                            Asset::ScalarOutcome(market_id, ScalarPosition::Short)
-                        }
-                    }
-                };
-
-                if remaining_amount > zero_balance {
-                    add_liqudity(remaining_amount, asset_in)?;
-                }
-            }
-
-            // Add additional liquidity for the base asset
-            let remaining_amount =
-                (amount_base_asset).saturating_sub(MinLiquidity::get().saturated_into());
-
-            if remaining_amount > zero_balance {
-                add_liqudity(remaining_amount, Asset::Ztg)?;
-            }
-
+            let weights_len = weights.len();
+            let _ = Self::deploy_swap_pool_for_market(origin, market_id, amount, weights)?;
             Ok(Some(
                 weight_bcs
-                    .saturating_add(T::WeightInfo::deploy_swap_pool_for_market(weight_len))
-                    .saturating_add(weight_pool_joins_and_sells)
+                    .saturating_add(T::WeightInfo::deploy_swap_pool_for_market(
+                        weights_len.saturated_into(),
+                    ))
                     .saturating_add(T::DbWeight::get().reads(2)),
```

Comment:
I think we don't need those reads anymore, because the two places that did storage reads were removed (all other reads are contained within functions which are benchmarked).
```suggestion
```